dist/ramda.js

```diff
@@ -4440,6 +4440,43 @@
         }, [], fns);
     };
 
+    // The algorithm used to handle cyclic structures is heavily
+    // inspired by underscore.js' implementation of isEqual
+    var _eqDeep = function _eqDeep(a, b, stackA, stackB) {
+        var idx = stackA.length;
+        while (--idx >= 0) {
+            if (stackA[idx] === a) {
+                return stackB[idx] === b;
+            }
+        }
+        stackA.push(a);
+        stackB.push(b);
+        var result;
+        switch (type(a)) {
+        case 'Array':
```

Comment:
@CrossEye Sorry I wasn't very clear.

``` javascript
if (type(a) === 'Date' || type(b) === 'Object') {
  if (type(a) === 'Date' && a.getTime() !== b.getTime()) {
    return false;
  }
  // now we check it as if it were an 'Object' ie check all attributes
  for (var key in a) {
    if (!(b[key] && eqDeep(a[key], b[key])) {
      return false;
    }
  }
}
```
This would test any added attributes